packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -252,10 +237,16 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     });
 
     this.paymentChannelClient.onChannelUpdated(async (channelState: ChannelState) => {
-      if (
-        channelState.channelId === wire.paidStreamingExtension.pseChannelId ||
-        channelState.channelId === wire.paidStreamingExtension.peerChannelId
-      ) {
+      const {pseChannelId, peerChannelId, peerAccount} = wire.paidStreamingExtension;
+
+      if (channelState.channelId === pseChannelId || channelState.channelId === peerChannelId) {
+        if (channelState.status === 'closed' || channelState.status === 'closing') {
+          wire.paidStreamingExtension.pseChannelId = null;
+          wire.paidStreamingExtension.peerChannelId = null;
+          log.info(
+            `PeerAccount ${peerAccount} - ChannelId ${channelState.channelId} Channel Closed||Closing`
+          );
```

Comment:
this does most of the work.
This updates the thing that gets checked [here](https://github.com/statechannels/monorepo/pull/1533/files#diff-a84153ed9f3ab2bd13c41bca43b584c7R190).
That triggers a new channel creation if the leecher makes new requests.